client/quantum_serverless/core/provider.py

```diff
@@ -494,6 +497,10 @@ def __init__(
         if host is None:
             raise QuantumServerlessException("Please provide `host` of gateway.")
 
+        version = version or os.environ.get(GATEWAY_PROVIDER_VERSION)
```

Comment:
This may cause using different versions `save_result` and the `provider`

Reply:
Sorry @akihikokuroda I'm not following you. Both resources use `GATEWAY_PROVIDER_VERSION` to obtain the value. Do you mean that how I'm letting the user provide the version through `args` the user could use different versions?

Reply:
Yes, if the user gives the arg with the different value from the value set in the env variable, does the `save_result` work correctly?

Reply:
Oh yes, now I see your point and you are totally right. My problem when I was writing this code was that (from my perspective) `save_result` should be refactored and live inside some of the upper classes that contains that information. I didn't do it in this PR because I thought there were too many changes and I would need some time to analyze if that method is as it is for some reason.

So I would like to open an issue instead to solve it in this PR maybe (I'm open to discuss it).